pyBumpHunter/bumphunter_1dim.py

```diff
@@ -726,27 +1044,89 @@ def bump_scan(self, data, bkg, is_hist: bool=False, do_pseudo: bool=True):
             if self.nworker > 1:
                 with thd.ThreadPoolExecutor(max_workers=self.nworker) as exe:
```

Comment:
Indeed, this kind of parallelization is not effective at all.
The problem is that it is very difficult to handle a shared memory between different python processes.
I tried some things, but I couldn't manage to make it work so far.
I hope some day I will find a solution to make it better.

Reply:
Is shared memory needed? Histograms are usually small in space, so this should be fine to be duplicated?

But I also wonder if the parallelisation is needed at all? Because in general, if avoidable, it's good to avoid it.

In fact, if, then I would suggest to go for [ray](https://github.com/ray-project/ray) maybe?

For now, I would suggest to completely remove this part. There is, AFAI see, no use case for this, because it will only slow down things without any speed improvement (using Threads) AFAIK. And once it's properly there, you can advertise it as an improvement.

(the more code, the more to maintain, the more to break... And adding code that is not functioning in the first place doesn't help anyone, right?)